docs/tutorial/from-zero-to-hero-write-your-first-kubernetes-charm/integrate-your-charm-with-postgresql.md

```diff
@@ -483,54 +483,44 @@ Now run `tox -e unit` to make sure all test cases pass.
 Now that our charm integrates with the PostgreSQL database, if there's not a database relation, the app will be in `blocked` status instead of `active`. Let's tweak our existing integration test `test_build_and_deploy` accordingly, setting the expected status as `blocked` in `ops_test.model.wait_for_idle`:
 
 ```python
-@pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest):
+import logging
+from pathlib import Path
+
+import jubilant
+import yaml
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path('./charmcraft.yaml').read_text())
+APP_NAME = METADATA['name']
+
+
+def test_build_and_deploy(charm: Path, juju: jubilant.Juju):
     """Build the charm-under-test and deploy it together with related charms.
 
-    Assert on the unit status before integration or configuration.
+    Assert on the unit status before any relations/configurations take place.
     """
     # Build and deploy charm from local source folder
-    charm = await ops_test.build_charm('.')
-    resources = {
-        'demo-server-image': METADATA['resources']['demo-server-image']['upstream-source']
-    }
+    resources = {'httpbin-image': METADATA['resources']['httpbin-image']['upstream-source']}
 
-    # Deploy the charm and wait for blocked/idle status.
-    # The app will not be in active status as this requires a database relation.
-    await asyncio.gather(
-        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME),
-        ops_test.model.wait_for_idle(
-            apps=[APP_NAME], status='blocked', raise_on_blocked=False, timeout=300
-        ),
-    )
+    # Deploy the charm and wait for active/idle status
```

Comment:
Why is it active/idle while previously it was blocked/idle? At this stage, Postgres is not deployed yet, right? So it should not be active?